front/lib/models/assistant/agent.ts

```diff
@@ -242,6 +242,31 @@ AgentConfiguration.belongsTo(User, {
   foreignKey: { name: "authorId", allowNull: false },
 });
 
+/**
+ * At time of writing, private agents can only be created from scratch.
+ * An existing agent that is not already private cannot become private.
+ * This is a framing decision that is enforced by the following hook.
+ */
+AgentConfiguration.addHook(
+  "beforeCreate",
+  "existing_agent_cannot_go_private",
+  async (agentConfiguration) => {
+    if (agentConfiguration.dataValues.scope === "private") {
+      // get the agent configuration with the same sId and maximum version
+      const existingAgentConfiguration = await AgentConfiguration.findOne({
```

Comment:
I think hoooks => pure data consistency but here we're going a bit further than that IMHO and this means we'll spread business logic in unsuspected places. Counter argument is that it will fail fast. But a similar check at the business logic layer would I think make our life easier in the future?